types/src/ledger/receipt.rs

```diff
@@ -4,21 +4,23 @@ use alloy_rpc_types::TransactionReceipt;
 use alloy_sol_types::SolValue;
 use serde::{Deserialize, Serialize};
 
-use super::{Transaction, log};
+use super::log;
 use crate::cryptography::{
     hash::{Hash, Hashable},
     merkle_tree::{MerkleBuilder, MerkleMultiProof, MerkleProof, Merkleizable, index_prefix},
-    signer::Signed,
 };
 
 #[derive(Clone, Serialize, Deserialize, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 pub struct Receipt {
     pub status: bool,
     pub actual_gas_used: u64,
+    pub max_fee_per_gas: u128,
     pub logs: Vec<Log>,
     pub logs_root: Hash,
-    pub tx: Signed<Transaction>,
+    pub tx_hash: Hash,
+    pub signer: Address,
+    pub to: Option<Address>,
     pub contract_address: Option<Address>,
 }
```

Comment:
Why include `max_fee_per_has`, `signer` and `to` fields in `Receipt`? They are part of the TX pointed to by `tx_hash` already. Additionally including them in the Receipt risks that they are assigned different values than the TX has. 

If this is to be able to implement `From<Receipt> for TransactionReceipt`, then perhaps this conversion shouldn't be possible with `From` impl anymore. Instead there could be `TryFrom<(Receipt, Transaction)>`, wdyt?